### PR TITLE
ui: Add filtering peer listing by State

### PR DESCRIPTION
**Pull request by johncowen**
### Description
Adds filtering peer listing by State

Screenshot 2022-07-07 at 12 45 39

Please note there are a few things that may be tweaked here moving forwards, i.e. specific icons etc. Right now we are just matching the previously used icons.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

/cc @LevelbossMike

### CODE REVIEW

File: ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs
Comment (natmegs) [inline ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs:87]: Def not blocking but I wonder if an enum could help here as a single source of truth for these states (to ensure they are in sync between here and predicates/peer for example)?

**Review by natmegs — APPROVED** [review]
I"m probably doing something wrong but I'm seeing some overlap in the peers table?
Screen Shot 2022-07-07 at 8 27 03 AM

**Comment by johncowen** [comment]
Hmm, that definitely doesn't look right 😆 thanks for spotting, let me take a look at this again brb

File: ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs
Comment (johncowen) [reply ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs:87]: Yeah agree completely, I tried to mostly centralise these in the model (I added an exported `schema`) in there, but as I'm mostly implementing (i.e. copy/pasting how its done elsewhere) I didn't put to much thought into doing anything extra. Agree its not a blocker for this, but let me have a think about this also 🤔

Comment (johncowen) [reply ui/packages/consul-ui/app/components/consul/peer/search-bar/index.hbs]: I reused our `schema.State.allowedValues` array here (imported from our model files), just waiting for tests to pass again, once thats done I'll merge away

**Pull request merged**
